bin/install

```diff
@@ -1,105 +1,149 @@
   local raw_version=$2
   local install_path=$3
-  local luaJIT_version=$(echo $raw_version | sed -e 's/--.*$//g')
-  local lua_rocks_version=$(echo $raw_version | sed -e "s/^.*--//g")
+  local luaJIT_version="${raw_version%%--*}"
+  local lua_rocks_version="${raw_version##*--}"
 
-  if [ "$TMPDIR" = "" ]; then
-    local tmp_download_dir=$(mktemp -d)
+  local tmp_download_dir
+  if [[ "${TMPDIR-}" == "" ]]; then
+    tmp_download_dir=$(mktemp -d)
   else
-    local tmp_download_dir=$TMPDIR
+    tmp_download_dir=$TMPDIR
   fi
 
-  local luaJIT_source_path=$(get_luaJIT_download_file_path $install_type $luaJIT_version $tmp_download_dir)
-  local lua_rocks_source_path=$(get_lua_rocks_download_file_path $install_type $lua_rocks_version $tmp_download_dir)
-  download_luaJIT_source $install_type $luaJIT_version $luaJIT_source_path
-  download_lua_rocks_source $install_type $lua_rocks_version $lua_rocks_source_path
+  # canonicalize the path.
+  # this has several benefits:
+  #  1. if the luajit/luarocks build-systems also does this (hint: they do),
+  #     then the path we use here will be guaranteed to be identical.
+  #  2. this removes silly stuff like //, /./ throughout the path,
+  #     and removes a trailing slash (e.g. macOS's $TMPDIR).
+  tmp_download_dir=$(realpath -- "$tmp_download_dir")
+
+  local luaJIT_source_path
+  luaJIT_source_path=$(get_luaJIT_download_file_path "$install_type" "$luaJIT_version" "$tmp_download_dir")
+  local lua_rocks_source_path
+  lua_rocks_source_path=$(get_lua_rocks_download_file_path "$install_type" "$lua_rocks_version" "$tmp_download_dir")
+  download_luaJIT_source "$install_type" "$luaJIT_version" "$luaJIT_source_path"
+  download_lua_rocks_source "$install_type" "$lua_rocks_version" "$lua_rocks_source_path"
 
   # running this in a subshell
   # because we don't want to disturb current working dir
   (
-    cd ${luaJIT_source_path}
+    local line
+
+    cd "${luaJIT_source_path}" || exit 1
+
+    local luaJIT_configure_options=()
+    while IFS='' read -r line; do
+      luaJIT_configure_options+=("$line")
+    done < <(construct_luaJIT_configure_options)
 
-    local luaJIT_configure_options="$(construct_luaJIT_configure_options)"
     # set in os_based_configure_options
     # we unset it here because echo-ing changes the return value of the function
     unset ASDF_PKG_MISSING
 
-    if [[ "$OSTYPE" = "darwin"* ]] && [[ -z "$MACOSX_DEPLOYMENT_TARGET" ]]; then
-      export MACOSX_DEPLOYMENT_TARGET=$(/usr/bin/sw_vers --productVersion)
+    if [[ "$OSTYPE" = "darwin"* ]] && [[ -z "${MACOSX_DEPLOYMENT_TARGET-}" ]]; then
+      MACOSX_DEPLOYMENT_TARGET=$(/usr/bin/sw_vers --productVersion)
+      export MACOSX_DEPLOYMENT_TARGET
     fi
 
-    echo "Building luaJIT with options: $luaJIT_configure_options"
-    make amalg $luaJIT_configure_options || exit 1
-    make install $luaJIT_configure_options || exit 1
+    printf "Building luaJIT with options:"
+    printf " %q" "${luaJIT_configure_options[@]}"
```

Comment:
The `%q` quotes the elements, making it clear if some argument actually does in fact have a space in it.

Before:

```
Building luaJIT with options: PREFIX=/some/path with a few spaces but this is another arg
```

vs now

```
Building luaJIT with options: 'PREFIX=/some/path with a few spaces' 'but this is another arg'
```